File: integrations/app/src/main/java/app/revanced/bilibili/account/model/AccountInfo.java
package app.revanced.bilibili.account.model;

import com.alibaba.fastjson.annotation.JSONField;

/* compiled from: BL */
/* loaded from: classes6.dex */
public class AccountInfo {
    public static final int ANSWER_STATUS_IN_PROGRESS = 2;
    public static final int ANSWER_STATUS_NOT = 1;
    public static final int SEX_TYPE_FEMALE = 2;
    public static final int SEX_TYPE_MALE = 1;
    public static final int SEX_TYPE_OTHER = 0;
    public static final int SILENCE_TYPE_EVER = 2;
    public static final int SILENCE_TYPE_NONE = 0;
    public static final int SILENCE_TYPE_NO_EVER = 1;

    @JSONField(name = "answer_status")
    public int answerStatus;

    @JSONField(name = "face")
    public String avatar;

    @JSONField(name = "birthday")
    public String birthday;

    @JSONField(name = "coins")
    public float coins;

    @JSONField(name = "email_status")
    public int emailStatus;

    @JSONField(name = "end_time")
    public long endTime;

    @JSONField(name = "has_face_nft")
    public boolean hasNft;

    @JSONField(name = "identification")
    public int identification;

    @JSONField(name = "in_reg_audit")
    public int inRegAudit;

    @JSONField(name = "is_jury")
    public boolean isJury;

    @JSONField(name = "is_senior_member")
    public int isSeniorMember;

    @JSONField(name = "set_birthday")
    public boolean isSetBirthday;

    @JSONField(name = "is_tourist")
    public int isTourist;

    @JSONField(name = "level")
    public int level;

    @JSONField(name = "mid")
    public long mid;

    @JSONField(name = "pin_prompting")
    public int pinPrompting;

    @JSONField(name = "rank")
    public int rank;

    @JSONField(name = "sex")
    public int sex;

    @JSONField(name = "sign")
    public String signature;

    @JSONField(name = "silence")
    public int silence;

    @JSONField(name = "tel_status")
    public int telStatus;

    @JSONField(name = "name")
    public String userName;

    @JSONField(name = "vip")
    public VipUserInfo vipInfo;

    public int getSilenceType() {
        if (silence == 0)
            return SILENCE_TYPE_NONE;
        return endTime == 0 ? SILENCE_TYPE_EVER : SILENCE_TYPE_NO_EVER;
    }

    public String getVipLabelTheme() {
        VipUserInfo vipUserInfo = vipInfo;
        if (vipUserInfo != null)
            return vipUserInfo.getLabelTheme();
        return null;
    }

    public boolean isFormalAccount() {
        return rank >= 10000;
    }

    public boolean isMobileVerified() {
        return telStatus > 0;
    }

    public boolean isRealName() {
        return identification == 1;
    }
}

File: integrations/app/src/main/java/app/revanced/bilibili/account/model/AccountInfo.kt
package app.revanced.bilibili.account.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class AccountInfo(
    @SerialName("answer_status")
    val answerStatus: Int = 0,
    @SerialName("face")
    val avatar: String = "",
    @SerialName("birthday")
    val birthday: String = "",
    @SerialName("coins")
    val coins: Float = 0f,
    @SerialName("email_status")
    val emailStatus: Int = 0,
    @SerialName("end_time")
    val endTime: Long = 0,
    @SerialName("has_face_nft")
    val hasNft: Boolean = false,
    @SerialName("identification")
    val identification: Int = 0,
    @SerialName("in_reg_audit")
    val inRegAudit: Int = 0,
    @SerialName("is_jury")
    val isJury: Boolean = false,
    @SerialName("is_senior_member")
    val isSeniorMember: Int = 0,
    @SerialName("set_birthday")
    val isSetBirthday: Boolean = false,
    @SerialName("is_tourist")
    val isTourist: Int = 0,
    @SerialName("level")
    val level: Int = 0,
    @SerialName("mid")
    val mid: Long = 0,
    @SerialName("pin_prompting")
    val pinPrompting: Int = 0,
    @SerialName("rank")
    val rank: Int = 0,
    @SerialName("sex")
    val sex: Int = 0,
    @SerialName("sign")
    val signature: String = "",
    @SerialName("silence")
    val silence: Int = 0,
    @SerialName("tel_status")
    val telStatus: Int = 0,
    @SerialName("name")
    val userName: String = "",
    @SerialName("vip")
    val vipInfo: VipUserInfo? = null
) {
    val silenceType: Int
        get() {
            if (silence == 0) return SILENCE_TYPE_NONE
            return if (endTime == 0L) SILENCE_TYPE_EVER else SILENCE_TYPE_NO_EVER
        }

    val vipLabelTheme: String
        get() = vipInfo?.labelTheme.orEmpty()

    val isFormalAccount: Boolean
        get() = rank >= 10000

    val isMobileVerified: Boolean
        get() = telStatus > 0

    val isRealName: Boolean
        get() = identification == 1

    companion object {
        const val ANSWER_STATUS_IN_PROGRESS: Int = 2
        const val ANSWER_STATUS_NOT: Int = 1
        const val SEX_TYPE_FEMALE: Int = 2
        const val SEX_TYPE_MALE: Int = 1
        const val SEX_TYPE_OTHER: Int = 0
        const val SILENCE_TYPE_EVER: Int = 2
        const val SILENCE_TYPE_NONE: Int = 0
        const val SILENCE_TYPE_NO_EVER: Int = 1
    }
}
